File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> um;
        for (int i=0; i<nums.size(); i++){
            if (um.find(nums[i]) != um.end()){
                return {i, um[nums[i]]};
            } else {
                um[target - nums[i]] = i;
            }
        }
        return {-1, -1};
    }
};

File: 0001-two-sum/0001-two-sum.java
import java.util.Map;
import java.util.HashMap;
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> hm = new HashMap<Integer, Integer>();
        for (int i=0; i<nums.length; i++){
            int find = target - nums[i];
            if (hm.get(find) != null){
                int[] ret = {hm.get(find), i};
                return ret;
            } else {
                hm.put(nums[i], i);
            }
        }
        int[] ret = {-1, -1};
        return ret;
    }
}